packages/sirius-web/frontend/sirius-web-application/src/omnibox/dialogs/SearchContentCommandDialogExtension.tsx

```diff
@@ -0,0 +1,238 @@
+import {
+  OmniboxCommandDialogComponentProps,
+  GQLGetOmniboxCommandsQueryVariables,
+  OmniboxAction,
+} from '@eclipse-sirius/sirius-components-omnibox';
+import { IconOverlay, useSelection } from '@eclipse-sirius/sirius-components-core';
+import { useOmniboxCommands } from '@eclipse-sirius/sirius-components-omnibox';
+import { useRef, useState } from 'react';
+import { SearchContentState } from './SearchContentCommandDialogExtension.types';
+
+import ChevronRightIcon from '@mui/icons-material/ChevronRight';
+import SubdirectoryArrowLeftIcon from '@mui/icons-material/SubdirectoryArrowLeft';
+import CircularProgress from '@mui/material/CircularProgress';
+import Dialog from '@mui/material/Dialog';
+import DialogContent from '@mui/material/DialogContent';
+import DialogTitle from '@mui/material/DialogTitle';
+import FormControl from '@mui/material/FormControl';
+import IconButton from '@mui/material/IconButton';
+import Input from '@mui/material/Input';
+import List from '@mui/material/List';
+import ListItem from '@mui/material/ListItem';
+import ListItemButton from '@mui/material/ListItemButton';
+import ListItemIcon from '@mui/material/ListItemIcon';
+import ListItemText from '@mui/material/ListItemText';
+import { useCurrentProject } from '../../views/edit-project/useCurrentProject';
+
+const useOmniboxStyles = makeStyles()((theme) => ({
+  omnibox: {
+    display: 'flex',
+    flexDirection: 'column',
+  },
+  omniboxPaper: {
+    position: 'fixed',
+    top: '10%',
+  },
+  omniboxInputArea: {
+    display: 'flex',
+    flexDirection: 'row',
+    alignItems: 'center',
+    paddingTop: theme.spacing(0),
+    paddingBottom: theme.spacing(0),
+    paddingLeft: theme.spacing(0),
+    paddingRight: theme.spacing(1),
+  },
+  omniboxIcon: {
+    minWidth: theme.spacing(4),
+    fontSize: '2.5rem',
+  },
+  omniboxFormControl: {
+    paddingTop: theme.spacing(1),
+    paddingBottom: theme.spacing(1),
+    paddingLeft: theme.spacing(0.5),
+    paddingRight: theme.spacing(0.5),
+    flexGrow: 1,
+  },
+  omniboxResultArea: {
+    maxHeight: '330px',
+    padding: theme.spacing(0),
+  },
+}));
+
+export const SearchContentCommandDialogExtension = ({ onClose }: OmniboxCommandDialogComponentProps) => {
+  const [state, setState] = useState<SearchContentState>({
+    queryHasChanged: true,
+  });
+
+  const { setSelection } = useSelection();
+  const { getOmniboxCommands, loading, data } = useOmniboxCommands();
+  const { project } = useCurrentProject();
+
+  const inputRef = useRef<HTMLInputElement>(null);
+  const listRef = useRef<HTMLUListElement>(null);
+
+  const onChange = () => {
+    if (!state.queryHasChanged) {
+      setState((prevState) => ({
+        ...prevState,
+        queryHasChanged: true,
+      }));
+    }
+  };
+
+  const sendQuery = (query: string) => {
+    setState((prevState) => ({
+      ...prevState,
+      queryHasChanged: false,
+    }));
+    const variables: GQLGetOmniboxCommandsQueryVariables = {
```

Comment:
This is the main issue with this component: it still queries the `Viewer#omniboxCommands` field, but it should probably query a dedicated endpoint (something like `EditingContext#content`. 
I think there is no reason to couple text-based search with the command API: it should be independent, and one of the ways to access it is through the command palette.